### PR TITLE
ocs-to-ocs: Add required env var in the API server deployment

### PR DESCRIPTION
**Pull request by iamniting**
STORAGE_CLUSTER_NAME and STORAGE_CLUSTER_UID is required for the garbage
collection of the storage consumer.

Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

File: controllers/storagecluster/provider_server.go [hunk 1]
```diff
@@ -202,6 +202,12 @@ func GetProviderAPIServerDeployment(instance *ocsv1.StorageCluster) *appsv1.Depl
 							Name:    "ocs-provider-api-server",
 							Image:   os.Getenv(providerAPIServerImage),
 							Command: []string{"/usr/local/bin/provider-api"},
+							Env: []corev1.EnvVar{
+								{
+									Name:  "WATCH_NAMESPACE",
+									Value: os.Getenv("WATCH_NAMESPACE"),
```
Comment (sp98) [inline controllers/storagecluster/provider_server.go:208]: There is a `GetWatchNamespace` method to get this namespace. It has error validations as well. Suggesting to use it here to get the `WATCH_NAMESPACE` value.

**Review by sp98 — CHANGES_REQUESTED** [review]

File: controllers/storagecluster/provider_server.go (hunk 1 above)
Comment (iamniting) [reply controllers/storagecluster/provider_server.go:216]: I am doing the same as a func for the other env, added this also to the list.

Comment (sp98) [reply controllers/storagecluster/provider_server.go:216]: so this approach ends up fetching the env variable data twice.  Once in `LookupEnv` and again in `Getenv`. Suggesting to have a single call to fetch the env variables.

**Review by sp98 — CHANGES_REQUESTED** [review]

File: controllers/storagecluster/provider_server.go (hunk 1 above)
Comment (iamniting) [reply controllers/storagecluster/provider_server.go:216]: Actually, I am doing that for a reason `createDeployment` func returns an error where I am checking it for the first time and returning an error if ENV does not exist. `GetProviderAPIServerDeployment` func does not return the error it just returns deployment only that's why not checking in that and I do not want to send ENVS as the argument.

Comment (sp98) [reply controllers/storagecluster/provider_server.go:216]: ok.

File: controllers/storagecluster/provider_server.go [hunk 2]
```diff
@@ -202,6 +210,20 @@ func GetProviderAPIServerDeployment(instance *ocsv1.StorageCluster) *appsv1.Depl
 							Name:    "ocs-provider-api-server",
 							Image:   os.Getenv(providerAPIServerImage),
 							Command: []string{"/usr/local/bin/provider-api"},
+							Env: []corev1.EnvVar{
+								{
+									Name:  "WATCH_NAMESPACE",
```
Comment (sp98) [inline controllers/storagecluster/provider_server.go:215]:
```suggestion
									Name:  util.WatchNamespaceEnvVar,
```

File: controllers/storagecluster/provider_server.go [hunk 3]
```diff
@@ -202,6 +210,20 @@ func GetProviderAPIServerDeployment(instance *ocsv1.StorageCluster) *appsv1.Depl
...
+									Value: os.Getenv("WATCH_NAMESPACE"),
```
Comment (sp98) [inline controllers/storagecluster/provider_server.go:216]:
```suggestion
									Value: os.Getenv(util.WatchNamespaceEnvVar),
```

File: controllers/storagecluster/provider_server.go [hunk 4]
```diff
@@ -85,8 +85,16 @@ func (o *ocsProviderServer) ensureDeleted(r *StorageClusterReconciler, instance
 
 func (o *ocsProviderServer) createDeployment(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
 
-	if _, ok := os.LookupEnv(providerAPIServerImage); !ok {
-		return fmt.Errorf("ENV var %s not found", providerAPIServerImage)
+	var finalErr error
+
+	for _, env := range []string{providerAPIServerImage, "WATCH_NAMESPACE"} {
```
Comment (sp98) [inline controllers/storagecluster/provider_server.go:90]:
```suggestion
	for _, env := range []string{providerAPIServerImage, util.WatchNamespaceEnvVar} {
```

**Review by sp98 — CHANGES_REQUESTED** [review]
nit

File: controllers/storagecluster/provider_server.go (hunk 2 above)
Comment (iamniting) [reply controllers/storagecluster/provider_server.go]: done

File: controllers/storagecluster/provider_server.go (hunk 3 above)
Comment (iamniting) [reply controllers/storagecluster/provider_server.go]: done

**Review by sp98 — APPROVED** [review]
/lgtm

**Review by agarwal-mudit — APPROVED** [review]
/lgtm

**Pull request merged**